332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/ConnectorUtils.java

```
@@ -204,8 +208,14 @@ public ConnectorDetails getConnectorDetails(NGAccess ngAccess, String connectorI
     if (isEmpty(connectorIdentifier)) {
       throw new CIStageExecutionException("Git connector is mandatory in case git clone is enabled");
     }
+    try {
+      connectorDetails = super.getConnectorDetails(ngAccess, connectorIdentifier);
+    } catch (ConnectorNotFoundException e) {
```

Catch all exceptions

---

Throw error as well